tests/branch_management/tests/conftest.py

```diff
@@ -2,26 +2,71 @@
 # Copyright 2024 Canonical, Ltd.
 #
 from pathlib import Path
+from typing import Optional
 
 import pytest
 import requests
 import semver
 
+STABLE_URL = "https://dl.k8s.io/release/stable.txt"
+RELEASE_URL = "https://dl.k8s.io/release/stable-{}.{}.txt"
 
-@pytest.fixture
-def upstream_release() -> semver.VersionInfo:
+
+def _upstream_release_exists(ver: semver.Version) -> Optional[semver.Version]:
```

Comment:
nit: I'd just name this `_upstream_release`. `_exists` IMHO signals that this function returns a bool